src/models/sssd_conf.py

```diff
@@ -1,34 +1,94 @@
-from .file import File
-from pathlib import Path
 from configparser import ConfigParser
+from pathlib import Path
+import os
+from shutil import copy2
 from SCAutolib import logger
+from SCAutolib.src import TEMPLATES_DIR, LIB_BACKUP
+from .file import File
 
 
 class SSSDConf(File):
-    _template = Path()
+    _template = Path(TEMPLATES_DIR, "sssd.conf")
     _sssd_conf_path = Path("/etc/sssd/sssd.conf")
-    _parser = ConfigParser()
-    # Needed to avoid parsing problems on inserting some 'specific' value,
-    # that cause an exception with default method
-    _parser.optionxform = str
-    with _sssd_conf_path.open() as f:
-        _parser.read_file(f)
+    _backup_original = Path(LIB_BACKUP, 'original_sssd.conf')
+    _backup_default = Path(LIB_BACKUP, 'default_sssd.conf')
+
+    @classmethod
+    def create(cls):
+        """
+        Creates sssd.conf file if not present in the system or update existing
+        sssd.conf file with values from template file provided by this package
+        and backup original file.
+        """
+        parser = ConfigParser()
+        # avoid problems with inserting some 'specific' values
+        parser.optionxform = str
+
+        # this method should be executed only once to create default config
+        # existing backup files indicates that this method was already executed
+        # in such case exception should be raised
+        if Path(cls._backup_original).is_file():
+            logger.error("Backup of original sssd.conf already exists")
+            raise FileExistsError('backup file exists')
+        if Path(cls._backup_default).is_file():
+            logger.error("Default sssd.conf backup already exists")
+            raise FileExistsError('backup file exists')
+
+        try:
+            with cls._sssd_conf_path.open() as config:
+                parser.read_file(config)
+            copy2(cls._sssd_conf_path, cls._backup_original)        # BACKUP
+        except FileNotFoundError:
+            logger.info(f"sssd.conf not present in {cls._sssd_conf_path}")
+            logger.info("Creating sssd.conf based on the template")
```

Comment:
Not an info message, but a warning. Consideration: to warn a user that sssd.conf that the user expects to be present, is not present, but we will create it on our own.